static/app/components/stateLayer.tsx

```diff
@@ -0,0 +1,67 @@
+import isPropValid from '@emotion/is-prop-valid';
+import styled from '@emotion/styled';
+
+import {defined} from 'sentry/utils';
+
+interface StateLayerProps extends React.HTMLAttributes<HTMLSpanElement> {
+  color?: string;
+  higherOpacity?: boolean;
+  isHovered?: boolean;
+  isPressed?: boolean;
+}
+
+const StateLayer = styled(
+  (props: StateLayerProps) => <span role="presentation" {...props} />,
+  {shouldForwardProp: p => typeof p === 'string' && isPropValid(p)}
+)`
+  position: absolute;
+  top: 50%;
+  left: 50%;
+  width: 100%;
+  height: 100%;
+  transform: translate(-50%, -50%);
+
+  box-sizing: content-box;
+  border-radius: inherit;
+  border: inherit;
+  color: ${p => (p.color ? p.theme[p.color] ?? p.color : 'currentcolor')};
+  background-color: currentcolor;
+  border-color: currentcolor;
+  pointer-events: none;
+
+  opacity: 0;
+
+  ${p =>
+    defined(p.isHovered)
+      ? `
+        *:not(.focus-visible) > & {
+          opacity: ${p.isHovered ? (p.higherOpacity ? 0.085 : 0.06) : 0};
```

Comment:
Should these numbers be in the theme? Not sure myself, just putting the question out there

Reply:
Currently these numbers are only used by `StateLayer` and I don't see them being useful elsewhere, so I think it's best that they're here. In the future we can move them to theme if there's another component that needs them.